rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/filter/TbDeviceTypeSwitchNodeTest.java

```diff
@@ -0,0 +1,123 @@
+import org.thingsboard.rule.engine.api.EmptyNodeConfiguration;
+import org.thingsboard.rule.engine.api.RuleEngineDeviceProfileCache;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.server.common.data.DeviceProfile;
+import org.thingsboard.server.common.data.id.CustomerId;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.id.EntityId;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.TbMsgMetaData;
+import org.thingsboard.server.common.msg.queue.TbMsgCallback;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TbDeviceTypeSwitchNodeTest {
+
+    TenantId tenantId;
+    DeviceId deviceId;
+    DeviceId deviceIdDeleted;
+    DeviceProfile deviceProfile;
+    TbContext ctx;
+    TbDeviceTypeSwitchNode node;
+    EmptyNodeConfiguration config;
+    TbMsgCallback callback;
+    RuleEngineDeviceProfileCache deviceProfileCache;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        tenantId = new TenantId(UUID.randomUUID());
+        deviceId = new DeviceId(UUID.randomUUID());
+        deviceIdDeleted = new DeviceId(UUID.randomUUID());
+
+        deviceProfile = new DeviceProfile();
+        deviceProfile.setTenantId(tenantId);
+        deviceProfile.setName("TestDeviceProfile");
+
+        //node
+        config = new EmptyNodeConfiguration();
+        node = new TbDeviceTypeSwitchNode();
+        node.init(ctx, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        //init mock
+        ctx = mock(TbContext.class);
+        deviceProfileCache = mock(RuleEngineDeviceProfileCache.class);
+        callback = mock(TbMsgCallback.class);
+
+        when(ctx.getTenantId()).thenReturn(tenantId);
+        when(ctx.getDeviceProfileCache()).thenReturn(deviceProfileCache);
+
+        doReturn(deviceProfile).when(deviceProfileCache).get(tenantId, deviceId);
+        doReturn(null).when(deviceProfileCache).get(tenantId, deviceIdDeleted);
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
+
+    @Test
+    void givenMsg_whenOnMsg_then_Fail() {
+        CustomerId customerId = new CustomerId(UUID.randomUUID());
+        assertThatThrownBy(() -> node.onMsg(ctx, getTbMsg(customerId))).isInstanceOf(RuntimeException.class);
```

Comment:
You should check the message of the RuntimeException. Otherwise, this may be any NPE or other instance of RuntimeException.